applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaMath.h

```diff
@@ -626,6 +626,15 @@ class /*__align__(4)*/ matrix3
                      x.z * v.x.z + y.z * v.y.z + z.z * v.z.z );
     }
 
+    __device__ matrix3<real> transpose(matrix3<real> v)
```

Comment:
I would have passed `v` by const reference. Same rules apply on CUDA than in c++. I am surprised it is not done for the other functions.
I am sure it does not impact the performances though.